casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -360,6 +362,18 @@ object Validate {
       } yield BlockStatus.invalidDeployShardId.asLeft[ValidBlock]
     }
 
+  // Shard ID of each deploy in the block should contain only ASCII characters
+  def deploysShardIdentifierOnlyAscii[F[_]: Monad: Log](b: BlockMessage): F[ValidBlockProcessing] =
+    if (b.body.deploys.forall(_.deploy.data.shardId.onlyAscii)) {
+      BlockStatus.valid.asRight[BlockError].pure
+    } else {
+      for {
+        _ <- Log[F].warn(
+              ignore(b, s"not for all deploys shard identifier contain only ASCII characters.")
+            )
+      } yield BlockStatus.invalidDeployShardId.asLeft[ValidBlock]
+    }
```

Comment:
We can do this check in `deploysShardIdentifier` for input shardId and existing check will check that all deploys contain the same _correct_ value.